doc/ws.md

```diff
@@ -316,12 +318,24 @@ This class represents a WebSocket. It extends the `EventEmitter`.
     Options given do not have any effect if parsed from the URL given with the
     `address` parameter.
 
+Create a new WebSocket instance.
+
 `perMessageDeflate` default value is `true`. When using an object, parameters
 are the same of the server. The only difference is the direction of requests.
 For example, `serverNoContextTakeover` can be used to ask the server to disable
 context takeover.
 
-Create a new WebSocket instance.
+`finishRequest` is called with arguments
+
+- `request` {http.ClientRequest}
+- `websocket` {WebSocket}
```

Comment:
Is this argument needed? The user already has access to the `WebSocket` instance returned by the constructor, no?

Reply:
`finishRequest` is called from the constructor, so the new WebSocket hasn't been returned yet

Reply:
Fair point, but is it needed? What would it be used for?

Reply:
I have no specific use-case, it just seemed polite to provide it to finishRequest just in case it's needed since it gets invoked before the caller has access to the websocket. It's not critical, but it takes no effort to pass it as argument while it would be annoying if it doesn't get passed and you do end up needing it.

Reply:
I think we can add it later if someone needs it.

Reply:
On second thought, it might be useful if someone want to call `websocket.close()` or `websocket.terminate()` synchronously.

Reply:
Or they may just want to keep track of the WebSocket for which they're doing asynchronous work for bookkeeping/debugging purposes. Passing it just seemed like a sensible thing to do.

Reply:
In the end I don't care enough to continue debating it, I'll remove the websocket argument.

Reply:
No, keep it. I quoted my comment, not yours.